pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -209,8 +203,12 @@ func (s *Service) handle(msg *message) error {
 	for !isNoOp(next) {
 		//TODO: Issue-578 is created to consider if we need to create connection ID at pre state level or no
 		s.sendMsgEvents(&service.StateMsg{
-			Type: service.PreState, Msg: msg.Msg, StateID: next.Name(),
-			Properties: createEventProperties("", "")})
+			ProtocolName: DIDExchange,
+			Type:         service.PreState,
+			Msg:          msg.Msg.Clone(),
```

Comment:
Msg is already pass by value. Any specific reason to Clone ?

Reply:
yeah, the payload can be modified by the user (bc Payload is slice and slice keeps a pointer to the array the hood e.g `msg.Payload[0] = 0x1`) and also `Header` is a pointer as well